src/Microsoft.DotNet.ImageBuilder/src/Commands/PublishImageInfoCommand.cs

```diff
@@ -42,17 +44,26 @@ public override Task ExecuteAsync()
             {
                 _loggerService.WriteSubheading("Cloning GitHub repo");
 
-                CloneOptions cloneOptions = new() { BranchName = Options.GitOptions.Branch };
-                cloneOptions.FetchOptions.CredentialsProvider = GetCredentials(Options.GitOptions.GitHubAuthOptions);
+                string authToken = await _octokitClientFactory.CreateGitHubTokenAsync(Options.GitOptions.GitHubAuthOptions);
+                CredentialsHandler credentials = GetCredentials(authToken);
 
-                using IRepository repo =_gitService.CloneRepository(
+                CloneOptions cloneOptions = new()
+                {
+                    BranchName = Options.GitOptions.Branch,
+                    FetchOptions = new FetchOptions
+                    {
+                        CredentialsProvider = credentials
+                    }
+                };
```

Comment:
CI hit: `/image-builder/src/Commands/PublishImageInfoCommand.cs(53,21): error CS0200: Property or indexer 'CloneOptions.FetchOptions' cannot be assigned to -- it is read only [/image-builder/src/Microsoft.DotNet.ImageBuilder.csproj]`

I don't have this open in an IDE, but I think this is the general approach given https://github.com/libgit2/libgit2sharp/blob/50d697881e857ea0c78d63cc7f23fece3f526852/LibGit2Sharp/CloneOptions.cs#L15:

```suggestion
                CloneOptions cloneOptions = new(new() { CredentialsProvider = credentials })
                {
                    BranchName = Options.GitOptions.Branch,
                };
```